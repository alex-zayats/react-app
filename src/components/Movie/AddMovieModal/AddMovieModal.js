import React from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'src/components/Common/Modal';
import { AddMovieForm } from 'src/components/Movie/AddMovieForm';

export function AddMovieModal({isModalOpened, closeModal}) {
  return (
    <Modal title="Really add?!" isModalOpened={isModalOpened} closeModal={closeModal}>
      <AddMovieForm></AddMovieForm>
    </Modal>
  );
}

AddMovieModal.propTypes = {
  isModalOpened: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired
}
