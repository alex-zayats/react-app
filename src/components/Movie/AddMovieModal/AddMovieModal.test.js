import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { AddMovieModal } from './AddMovieModal';

jest.mock('src/components/Common/Modal', () => {
  return { Modal: (props) => <div className="modal">{props.title}</div> };
});

jest.mock('src/components/Movie/AddMovieForm', () => {
  return { AddMovieForm: () => <div className="form"></div> };
});

describe('AddMovieModal component', () => {
  it('should render component', () => {
    const closeModal = () => false;
    const { asFragment } = render(<AddMovieModal isModalOpened={true} closeModal={closeModal}></AddMovieModal>);
    expect(asFragment).toMatchSnapshot();
  });
});
