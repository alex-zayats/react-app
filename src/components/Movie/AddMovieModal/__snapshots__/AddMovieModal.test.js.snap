// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieModal component should render component 1`] = `[Function]`;
